### PR TITLE
fix: add probability guard to CopyPaste in flip mode

### PR DESCRIPTION
**Pull request by Rahulbiradar9**
This PR adds the missing image-level probability guard `random.uniform(0, 1) > self.p` to the `CopyPaste` augmentation class when operating in `"flip"` mode (the default mode).

### Details
Previously, in `"flip"` mode, the `CopyPaste` transform was unconditionally applied to every image containing segments, making the effective image-level probability 1.0. Instead of behaving as a probability of applying the transformation, the `copy_paste` hyperparameter (`self.p`) was only used as a slicing fraction to select a portion of the candidate objects.

This PR adds the probability check to `"flip"` mode (matching the existing `"mixup"` mode logic), so that `copy_paste` represents the probability of applying the transformation to a given image.

### Verification
- Formatted and linted the changes using `ruff`.
- Ran the pretrained training unit tests successfully:
  ```bash
  pytest tests/test_python.py::test_train_pretrained -v

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds a probability guard to `CopyPaste` when operating in flip mode.

### 📊 Key Changes
- Samples a random value and skips augmentation when it exceeds the configured `self.p` probability.
- Applies the guard before generating flip parameters or modifying images and instances.
- Leaves existing behavior unchanged for empty segments and zero-probability configurations.

### 🎯 Purpose & Impact
- Ensures flip-mode CopyPaste respects its configured augmentation probability.
- Prevents unintended augmentation on samples that should be skipped.
- Improves consistency with probabilistic augmentation expectations while keeping the change narrowly scoped.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Rahulbiradar9, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

🤖 This is an automated message, and an engineer will assist with reviewing your contribution.

File: ultralytics/data/augment.py [hunk 1]
```diff
@@ -1919,6 +1919,8 @@ def __call__(self, labels: dict[str, Any]) -> dict[str, Any]:
         if len(labels["instances"].segments) == 0 or self.p == 0:
             return labels
         if self.mode == "flip":
+            if random.uniform(0, 1) > self.p:
```
Comment (UltralyticsAssistant) [inline ultralytics/data/augment.py]: 💡 **MEDIUM**: This gate makes `self.p` control two independent stages: whether flip CopyPaste runs here and how many eligible instances are selected later (`selected = indexes[: round(self.p * n)]`). Consequently, at `p=0.5` an image is processed only half the time and copies only half its candidates when it is processed; for a single eligible instance and any `p < 0.5`, `round(self.p * n)` is zero, so the image is never actually augmented even when this guard passes. That does not implement the stated image-level probability semantics. Once this guard is introduced, select all eligible candidates in flip mode (and decide whether the analogous mixup behavior should be updated for consistency).

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The new guard is correctly placed before expensive CopyPaste work, but the existing `self.p`-based instance slicing remains and compounds the probability, so the implementation does not deliver the PR's stated image-level probability behavior.

💬 Posted 1 inline comment

File: ultralytics/data/augment.py (hunk 1 above)
Comment (Rahulbiradar9) [reply ultralytics/data/augment.py]: I've updated the implementation to fix the double-stage control issue:
1. Added the proper image-level probability guard `random.random() >= self.p` to flip mode (consistent with `BaseMixTransform.__call__`).
2. Removed `self.p` from the instance selection logic in `CopyPaste.get_params()` entirely. All eligible non-overlapping candidates are now selected and pasted, consistent across both `flip` and `mixup` modes of `CopyPaste`.

**Review by glenn-jocher — APPROVED** [review]
Independently reproduced on the live head: flip-mode CopyPaste with one eligible object and p=0.5 never changes the image on main because round(p * n) is zero. This head moves probability to the image-level owner and selects all eligible candidates, matching the documented contract with a minimal net-negative diff. Full diff LGTM.
